types/src/json_pretty_printer.rs

```diff
@@ -23,40 +23,46 @@ where
     serde_json::to_string_pretty(&json_value)
 }
 
-fn shorten_string_field(value: &mut Value) {
-    match value {
-        Value::String(string) => {
-            let mut ranges_to_remove = Vec::new();
-            let mut start_index = 0;
-            let mut contiguous_hex_count = 0;
-
-            // Record all large hex-strings' start positions and lengths.
-            for (index, char) in string.char_indices() {
-                if char.is_ascii_hexdigit() {
-                    if contiguous_hex_count == 0 {
-                        // This is the start of a new hex-string.
-                        start_index = index;
-                    }
-                    contiguous_hex_count += 1;
-                } else if contiguous_hex_count != 0 {
-                    // This is the end of a hex-string: if it's too long, record it.
-                    if contiguous_hex_count > MAX_STRING_LEN {
-                        ranges_to_remove.push((start_index, contiguous_hex_count));
-                    }
-                    contiguous_hex_count = 0;
-                }
+fn find_hex_strings_longer_than(s: &str, max_len: usize) -> Vec<(usize, usize)> {
```

Comment:
Can you add a comment to the method? The return type is very cryptic and could mean a lot of different things.

Reply:
I'd prefer to add a self-explaining type instead of a comment, like so:
```rust
struct SubstringSpec {
    start_index: usize,
    length: usize
}

fn find_hex_strings_longer_than(s: &str, max_len: usize) -> Vec<SubstringSpec> {...
```

What do you think?

Reply:
I certainly prefer the struct to a tuple.

Reply:
I think the function signature reads well 👍🏼 .
The self documenting struct is nicer to read IMO.

Reply:
Why not both? If that function was in a library (and it seems to be generic enough to potentially be there), it would have a comment. I think people get uncle Bob's "great" advices too serious.

Reply:
I must admit, I'd be inclined to add a doc comment to the function myself, but I don't feel strongly enough to insist since the function is well-named, and if the return type is changed too, it becomes pretty self-documenting.

Your call @rafal-ch - go for what you prefer.

Reply:
I'll do both :)
Thanks for the insights.

Reply:
Resolved in 3cfc1105f6c4e1628dccaffb9d6cf2c9446f85a4